File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Deque;
import java.util.LinkedList;

class Solution {
    boolean solution(String s) {
        boolean answer = false;

        Deque<Character> deque = new LinkedList<>();
        
        char[] charArray = s.toCharArray();

        for (char c : charArray) {
        	
        	if(c == '(') {
        		deque.add(c);
        	}  else if(c == ')') {
        		if(deque.size() == 0) {
        			return false;
        		} else {
        			deque.poll();
        		}
        	}
		}
        
        if(deque.size() == 0) {
        	return true;
        }

        return answer;
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(text){

    let result = 0
    let answer = true
    
    for (let obj of text) {
      
        if (obj === '(') {
            result++
        } else {
            result--
            if (result < 0) return answer = false
        }

    }

    return result === 0 && answer

}